### PR TITLE
OASIS StyleGAN - S4641799

### PR DESCRIPTION
**Pull request by eco9898**
# Task
Create a generative model of one of the OASIS brain, ADNI brain or the OAI AKOA knee data set (see Appendix for links) using either a variant of StyleGAN /StyleGAN2 or Stable Diffusion that has a “reasonably clear image”.

In this project, Stable Diffusion was experimented with, but StyleGAN was the final implementation on the OASIS brain dataset. Stable Diffusion wasn't used due to a lack of familiarity with the model type.
# Project Files
- `dataset.py`: Data loader for the OASIS brain dataset, transforms the data for use with the model.
- `modules.py`: The StyleGAN networks as well as the layers and blocks they use.
- `predict.py`: Generates a batch of fake images that can be compared to the original dataset.
- `train.py`: Training process for the StyleGAN networks, will also generate example images and graph the loss function.
- `utils.py`: Functions used by the other scripts and global variables.
# Results
By epoch 150 I consider the model a success, however, it was starting to generate repeated and similar images. Training the model any longer would most likely cause the model to become overtrained and collapse the training simulation.
1697679243 0313046_predicted

### CODE REVIEW

**Comment by WeiDaiwow** [comment]
# Document
 
- Title: 
	Good
- Address problem:
	Problem mentioned, but no reasons were provided for why Style GAN is good for this task.
- Working Principle:
	Good
- Figure visualization
	Good
- dependency
	Good
- inputs and outputs description:
	input visualization not provided.
	output visualization provided.
- training plot:
	loss plot only contains single loss. Is this discriminator or generator? Should plot both
- readme using markdown:
	Good
- preprocessing description:
	Not mentioned
- Description of usage:
	Good

Suggestions:
1. the link in noise injection is broken 


# Check the files:
- modules.py
	no comments provided
- dataset.py
	not enough comments
- train.py
	not enough comments
- predict.py
	Good



# Overall:

- solve the problem:
	Yes
- TF/torch implementation:
	Yes
- Design quality:
	Good
- code comments:
	not enough comments for modules.py, dataset.py and train.py
- Difficulty: 
	styleganv2 -hard

**Comment by eco9898** [comment]
New commits were added to address:
1. Noise injection image not appearing
2. Adding noise input examples to the readme
3. Improving and adding comments to the python files
4. Updating the Readme content to include justification for why StyleGAN was chosen
5. Updating the Readme content to include preprocessing done on training images 
6. Updating the readme content to fix minor formatting and style issues

If any more changes are required please let me know, but I believe this addresses all the issues raised in the feedback. @WeiDaiwow

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting minimal -1
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in InCorrect Branch) -2
Feedback required, fix branch to correct branch
Request Description good

**Comment by shakes76** [comment]
Cannot merge because of gitignore file and incorrect name of folder, not stable diffusion but Style GAN, please amend. Does not affect grade only the merge.

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks granted.

**Comment by eco9898** [comment]
Sorry, I didn't receive a notification about these comments and as semester had ended was no longer actively checking for comments. I will work on these changes now. @shakes76 and @wangzhaomxy

**Comment by eco9898** [comment]
Addressed issues relating to the pull request's branch

**Comment by eco9898** [comment]
Feedback complete and an issue was raised on edstem: https://edstem.org/au/courses/13207/discussion/1726226, which has now been answered

**Comment by wangzhaomxy** [comment]
Feedback attempt and feedback marks granted +2.
